build.psm1

```diff
@@ -2194,9 +2194,9 @@ function Start-PSBootstrap {
     }
 }
 
-function Get-LatestInstalledSDK {
+function Get-DotNetVersionInPath {
     Start-NativeExecution -sb {
-        dotnet --list-sdks | Select-String -Pattern '\d*.\d*.\d*(-\w*\.\d*.\d*.\d*)?' | ForEach-Object { [System.Management.Automation.SemanticVersion]::new($_.matches.value) } | Sort-Object -Descending | Select-Object -First 1
+        dotnet --version
```

Comment:
I don't think this will work. When running within the repo folder, `global.json` will be honored, so `dotnet --version` will returns messages like the following when the dotnet version doesn't match:
```
PS:15> dotnet --version
Could not execute because the application was not found or a compatible .NET SDK is not installed.
Possible reasons for this include:
  * You intended to execute a .NET program:
      The application '--version' does not exist.
  * You intended to execute a .NET SDK command:
      A compatible installed .NET SDK for global.json version [7.0.100-preview.2.22153.17] from [C:\arena\source\PowerShell\global.json] was not found.
      Install the [7.0.100-preview.2.22153.17] .NET SDK or update [C:\arena\source\PowerShell\global.json] with an installed .NET SDK:
        5.0.402 [C:\Program Files\dotnet\sdk]
        6.0.202 [C:\Program Files\dotnet\sdk]
```

Reply:
I'm afraid we still need `dotnet --list-sdks`, but instead of returning the highest version, we need to check if the returned versions contain the required version.